gadopt/utility.py

```diff
@@ -7,13 +7,14 @@
 from firedrake import outer, ds_v, ds_t, ds_b, CellDiameter, CellVolume, dot, JacobianInverse
 from firedrake import sqrt, Function, FiniteElement, TensorProductElement, FunctionSpace, VectorFunctionSpace
 from firedrake import as_vector, SpatialCoordinate, Constant, max_value, min_value, dx, assemble, tanh
-from firedrake import op2, VectorElement, DirichletBC, utils, interpolate
+from firedrake import op2, VectorElement, DirichletBC, utils, interpolate, conditional
 from firedrake.ufl_expr import extract_unique_domain
 import ufl
 import time
 from ufl.corealg.traversal import traverse_unique_terminals
 from firedrake.petsc import PETSc
 from mpi4py import MPI
+from numbers import Number
```

Comment:
Is this necessary, or can we use `float` instead?

Reply:
done